cylc/flow/print_tree.py

```diff
@@ -62,18 +64,28 @@ def print_tree(tree, padding, use_unicode=False, prefix='', labels=None,
         result = pp + item
         line = result + ' ' + padding[len(result):]
         if isinstance(tree[item], dict):
-            print(line)
-            print_tree(tree[item], padding, use_unicode, pprefix, labels, eq)
+            ret.append(line)
+            ret.extend(
+                print_tree(
+                    tree[item],
+                    padding,
+                    use_unicode,
+                    pprefix,
+                    labels,
+                    eq
+                )
+            )
         else:
             if labels:
                 if item in labels:
                     reason = labels[item][1]
-                    print(line, '...', reason)
+                    ret.append(f'{line} ... {reason}')
                 else:
-                    print(line)
+                    ret.append(line)
             else:
                 if eq:
                     joiner = '= '
                 else:
                     joiner = ''
-                print(line + joiner + str(tree[item]))
+                ret.append(f'{line}{joiner}{tree[item]}')
+    return ret
```

Comment:
https://github.com/cylc/cylc-flow/blob/c56a1b825c63b89196872e342500f43eb7b6b9bb/cylc/flow/config.py#L1398
https://github.com/cylc/cylc-flow/blob/c56a1b825c63b89196872e342500f43eb7b6b9bb/cylc/flow/scripts/cylc_broadcast.py#L271
 ^^^ These calls won't do anything now, I think they were expecting a printed tree rather than a built tree. I see you have fixed the call on scripts/cylc-scan.py `  ret = print_tree(tree, '', sort=False, use_unicode=True)` . Does the same fix have to happen in the above files?

Reply:
Good spot, I hadn't realised there were other uses, this probably explains the broadcast failure!

Reply:
The clue was in the name!  `print_tree`